dracclient/utils.py

```diff
@@ -396,21 +423,26 @@ def set_settings(settings_type,
         return build_return_dict(
             None,
             resource_uri,
+            include_commit_required=include_commit_required,
             is_commit_required_value=False,
-            is_reboot_required_value=constants.RebootRequired.false)
+            is_reboot_required_value=constants.RebootRequired.false,
+            commit_required_value=False)
 
     selectors = {'CreationClassName': cim_creation_class_name,
                  'Name': cim_name,
                  'SystemCreationClassName': 'DCIM_ComputerSystem',
                  'SystemName': 'DCIM:ComputerSystem'}
-
     properties = {'Target': target,
-                  'AttributeName': attrib_names,
                   'AttributeValue': [new_settings[attr] for attr
                                      in attrib_names]}
-
+    if settings_type == 'RAID':
+        properties['AttributeName'] = [current_settings[attr].name for
+                                       attr in attrib_names]
+    else:
+        properties['AttributeName'] =  attrib_names
     doc = client.invoke(resource_uri, 'SetAttributes',
                         selectors, properties,
                         wait_for_idrac=wait_for_idrac)
 
-    return build_return_dict(doc, resource_uri)
+    return build_return_dict(doc, resource_uri,
+                             include_commit_required=include_commit_required)
```

Comment:
We should never include commit_required in the results.  This parameter was deprecated and removed from python-dracclient.